File: ElecSysVC-Fronted-main/src/app/contratos/ContratosClases/contrato-entidad.ts
export class ContratoEntidad {
    id_contrato!: number;
    id_trabajador!: number;
    sueldo!: number;
    fecha_expedicion!: Date | null;
    fecha_iniciacion!: Date | null;
    id_trabajador_encargado!: number;
    cargo!: string;
    tipo_contrato!: string;
    estado!: string;
    fecha_nacimiento!: Date | null;
    lugar_nacimiento!: string;
    edad!: number;
    estadoCivil!: string;
}

export class TrabajadorEntidad{
    id_trabajador!: number;
    nombre!: string;
    telefono!: string;
    direccion!: string;
    correo!: string;
    tipo_usuario!: string;
    password!: string;
    estado!: string;
}

File: ElecSys-main/ElecSys/src/main/java/co/edu/unbosque/ElecSys/contrato/entidadCon/ContratoEntidad.java
package co.edu.unbosque.ElecSys.contrato.entidadCon;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;

@Entity
@Table (name = "contrato")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class ContratoEntidad {

    @Id
    @GeneratedValue(
            strategy = GenerationType.SEQUENCE,
            generator = "contrato_seq"
    )
    @SequenceGenerator(
            name = "contrato_seq",
            sequenceName = "seq_contrato_id",
            allocationSize = 1
    )
    @Column(name = "id_contrato")
    private Integer id_contrato;

    @Column(name = "id_trabajador")
    private int id_trabajador;

    @Column(name = "sueldo")
    private BigDecimal sueldo;

    @Column(name = "fecha_expedicion")
    private LocalDate fecha_expedicion;

    @Column(name = "fecha_iniciacion")
    private LocalDate fecha_iniciacion;

    @Column(name = "id_trabajador_encargado")
    private int id_trabajador_encargado;

    @Column(name = "cargo")
    private String cargo;

    @Column(name = "tipo_contrato")
    private String tipo_contrato;

    @Column(name = "estado")
    private String estado;

    @Column(name = "fecha_nacimiento")
    private LocalDate fecha_nacimiento;

    @Column(name = "lugar_nacimiento")
    private String lugar_nacimiento;

    @Column(name = "edad")
    private int edad;

    @Column(name = "estadocivil")
    private String estadoCivil;
}
